MultiKueue JobSet managedBy (#1870)
kubernetes-sigs/kueue: site/content/en/docs/concepts/multikueue.md

There is an ongoing effort to overcome these limitations by adding the possibility to disable the reconciliation of some jobs by the Kubernetes `batch/Job` controller. Details in `kubernetes/enhancements` [KEP-4368](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-label-for-batch-jobs#readme).

### JobSet

> MultiKueue expects the JobSets, submitted to a ClusterQueue using a MultiKueue admission check, to have `spec.managedBy` set to `kueue.x-k8s.io/multikueue`. The JobSet `managedBy` field is available since JobSet v0.5.0.

Author: done + added some "or else ..."

> Since unsuspending a JobSet in the manager cluster will lead to its local execution and updating the status of a local JobSet could conflict with its main controller, MultiKueue expects the JobSets submitted to a ClusterQueue using it to have `alpha.jobset.sigs.k8s.io/managed-by` label set to `kueue`.

Reviewer: ```suggestion
MultiKueue expects the JobSets submitted to a ClusterQueue to have the `spec.managedBy` field set to `kueue.x-k8s.io/multikueue`.
```
Once we migrate to use the field, I would suggest `kueue.x-k8s.io/multikueue` as the reserved value, by analogy to k8s, where the value is `kubernetes.io/job-controller`.

> Since unsuspending a JobSet in the manager cluster will lead to its local execution and updating the status of a local JobSet could conflict with its main controller, MultiKueue expects the JobSets submitted to a ClusterQueue using it to have `spec.managedBy` set to `kueue.x-k8s.io/multikueue`. The JobSet `managedBy` field is available since JobSet v0.5.0.

Reviewer: ```suggestion
MultiKueue expects the JobSets, submitted to a ClusterQueue using the MultiKueue amission check, to have `spec.managedBy` set to `kueue.x-k8s.io/multikueue`. The JobSet `managedBy` field is available since JobSet v0.5.0.
```
